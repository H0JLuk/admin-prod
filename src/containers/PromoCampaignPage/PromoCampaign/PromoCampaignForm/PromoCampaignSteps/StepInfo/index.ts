import StepInfo from './StepInfo';

export type { StepInfoProps } from './StepInfo';

export default StepInfo;
