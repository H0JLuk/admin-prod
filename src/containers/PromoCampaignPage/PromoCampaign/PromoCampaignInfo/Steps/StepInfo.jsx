import React, { useEffect, useState } from 'react';
import cn from 'classnames';
import { Col, Radio, Row, Switch } from 'antd';
import { getClientAppList } from '../../../../../api/services/clientAppService';
import { getAppCode } from '../../../../../api/services/sessionService';
import promoCodeTypes from '../../../../../constants/promoCodeTypes';

import style from '../PromoCampaignInfo.module.css';

const EXCURSION = 'Экскурсия';
const GIFT = 'Подарок';
const CATEGORY = 'Категории';
const NO_CATEGORY_LABEL = 'Нет выбранных категорий';
const URL_PROMO_CAMPAIGN = 'Ссылка на страницу промо-кампании';
const EMPTY_URL_PROMO_CAMPAIGN = 'Нет ссылки';
const TYPE_PROMO_CAMPAIGN = 'Тип промо-кампании';
const TYPE_PROMO_CODE = 'Тип промокода';
const ACTIVE_PERIOD = 'Период действия';
const SHOW_PROMO_CAMPAIGN = 'Витрины, в которых отображается промо-кампания';
const DZO = 'ДЗО';
const STATUS = 'Активность промо-кампании';
const URL_SOURCE_LABEL = 'Источник ссылки';
const URL_SOURCE_DZO_LABEL = 'ДЗО';
const SHOW_GO_TO_LINK_LABEL = 'Отображать кнопку "Перейти на сайт"';
const SHOW_ONLY_BUNDLE = 'Отображать только в составе бандла';
const EXTERNAL_ID_LABEL = 'Внешний ID';
const EMPTY_EXTERNAL_ID_LABEL = 'Внешний ID не выбран';
const URL_SOURCE_PROMO_CAMPAIGN_LABEL = 'Промо-кампания';

const STATUS_TYPE = {
    ACTIVE: 'Активная',
    INACTIVE: 'Неактивная',
};

const StepInfo = ({ promoCampaign }) => {

    const [clientApp, setClientApp] = useState();
    const appCode = getAppCode();

    useEffect(() => {
        (async () => {
            const { clientApplicationDtoList = [] } = await getClientAppList();
            const app = clientApplicationDtoList.find(({ code }) => code === appCode);
            setClientApp(app);
        })();
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);

    return (
        <div className={ style.containerStep }>
            <div className={ style.container }>
                <Row gutter={ [24, 40] }>
                    <Col span={ 24 }>
                        <div className={ style.infoTitle }>{ CATEGORY }</div>
                        <div className={ style.infoText }>
                            { promoCampaign.categoryList?.length
                                ? promoCampaign.categoryList.map(({ categoryName }) => categoryName).join(', ')
                                : <i>{ NO_CATEGORY_LABEL }</i>
                            }
                        </div>
                    </Col>

                    <Col span={ 8 }>
                        <div className={ style.infoTitle }>{ URL_PROMO_CAMPAIGN }</div>
                        <div className={ style.webLink }>
                            { promoCampaign.webUrl || <i>{ EMPTY_URL_PROMO_CAMPAIGN }</i> }
                        </div>
                    </Col>

                    <Col span={ 8 }>
                        <div className={ style.infoTitle }>{ URL_SOURCE_LABEL }</div>
                        <div className={ style.flexRow }>
                            <Radio
                                className={ style.checkbox }
                                checked={ promoCampaign?.settings?.priorityOnWebUrl !== true }
                                disabled
                            >
                                { URL_SOURCE_DZO_LABEL }
                            </Radio>
                            <Radio
                                className={ style.checkbox }
                                checked={ promoCampaign?.settings?.priorityOnWebUrl === true }
                                disabled
                            >
                                { URL_SOURCE_PROMO_CAMPAIGN_LABEL }
                            </Radio>
                        </div>
                    </Col>

                    <Col span={ 8 } className={ style.switchContainer }>
                        <div className={ style.switch }>
                            <div className={ style.infoTitle }>{ SHOW_GO_TO_LINK_LABEL }</div>
                            <Switch disabled checked={ promoCampaign.settings?.alternativeOfferMechanic } />
                        </div>
                    </Col>

                    <Col span={ 8 }>
                        <div className={ style.infoTitle }>{ DZO }</div>
                        <div className={ style.infoText }>
                            { promoCampaign.dzoName }
                        </div>
                    </Col>

                    <Col span={ 8 }>
                        <div className={ style.infoTitle }>{ ACTIVE_PERIOD }</div>
                        <div className={ style.infoText }>
                            { promoCampaign.startDate } - { promoCampaign.finishDate }
                        </div>
                    </Col>

                    <Col span={ 8 } className={ style.switchContainer }>
                        <div className={ style.switch }>
                            <div className={ style.infoTitle }>{ SHOW_ONLY_BUNDLE }</div>
                            <Switch disabled checked={ promoCampaign.standalone } />
                        </div>
                    </Col>

                    <Col span={ 8 }>
                        <div className={ style.infoTitle }>{ TYPE_PROMO_CODE }</div>
                        <div className={ style.infoText }>
                            { promoCodeTypes[promoCampaign.promoCodeType].label }
                        </div>
                    </Col>

                    <Col span={ 8 }>
                        <div className={ style.infoTitle }>{ EXTERNAL_ID_LABEL }</div>
                        <div className={ style.infoText }>
                            { promoCampaign.externalId || <i>{ EMPTY_EXTERNAL_ID_LABEL }</i> }
                        </div>
                    </Col>

                    <Col span={ 8 }>
                        <div className={ style.infoTitle }>{ STATUS }</div>
                        <div className={ cn(style.status, { [style.danger]: !promoCampaign.active }) }>
                            { promoCampaign.active ? STATUS_TYPE.ACTIVE : STATUS_TYPE.INACTIVE }
                        </div>
                    </Col>
                </Row>
            </div>

            <div className={ style.infoDetail }>
                <Row gutter={ [16, 16] }>
                    <Col span={ 12 }>
                        <div className={ style.container }>
                            <div className={ style.infoTitle }>{ SHOW_PROMO_CAMPAIGN }</div>
                            <div className={ style.infoText }>
                                { clientApp && clientApp.displayName }
                            </div>
                        </div>
                    </Col>
                    <Col span={ 12 }>
                        <div className={ style.container }>
                            <div className={ style.infoTitle }>{ TYPE_PROMO_CAMPAIGN }</div>
                            <div className={ style.flex }>
                                <Radio
                                    className={ style.checkbox }
                                    checked={ promoCampaign.type === 'NORMAL' }
                                    disabled
                                >
                                    { EXCURSION }
                                </Radio>
                                <Radio
                                    className={ style.checkbox }
                                    checked={ promoCampaign.type === 'PRESENT' }
                                    disabled
                                >
                                    { GIFT }
                                </Radio>
                            </div>
                        </div>
                    </Col>
                </Row>
            </div>
        </div>
    );
};

export default StepInfo;