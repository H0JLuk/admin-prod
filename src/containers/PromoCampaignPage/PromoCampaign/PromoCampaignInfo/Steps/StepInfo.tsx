import React, { useEffect, useState } from 'react';
import cn from 'classnames';
import { Col, Radio, Row, Switch } from 'antd';
import { getClientAppInfo } from '@apiServices/clientAppService';
import { getAppCode } from '@apiServices/sessionService';
import promoCodeTypes from '@constants/promoCodeTypes';
import behaviorTypes from '@constants/behaviorTypes';
import { BANNER_TYPE } from '@constants/common';
import { ClientAppDto, PromoCampaignDto } from '@types';

import style from '../PromoCampaignInfo.module.css';

type IStepInfoProps = {
    promoCampaign: PromoCampaignDto;
};

const EXCURSION = 'Экскурсия';
const GIFT = 'Подарок';
const CATEGORY = 'Категории';
const NO_CATEGORY_LABEL = 'Нет выбранных категорий';
const URL_PROMO_CAMPAIGN = 'Ссылка на страницу промо-кампании';
const EMPTY_URL_PROMO_CAMPAIGN = 'Нет ссылки';
const TYPE_PROMO_CAMPAIGN = 'Тип промо-кампании';
const TYPE_PROMO_CODE = 'Тип промокода';
const ACTIVE_PERIOD = 'Период действия';
const SHOW_PROMO_CAMPAIGN = 'Витрина, в которой отображается промо-кампания';
const DZO = 'ДЗО';
const STATUS = 'Активность промо-кампании';
const URL_SOURCE_LABEL = 'Источник ссылки';
const URL_SOURCE_DZO_LABEL = 'ДЗО';
const SHOW_GO_TO_LINK_LABEL = 'Отображать кнопку "Перейти на сайт"';
const SHOW_ONLY_IN_BUNDLE = 'Отображать только в составе бандла';
const EXTERNAL_ID_LABEL = 'Внешний ID';
const EMPTY_EXTERNAL_ID_LABEL = 'Внешний ID не выбран';
const DETAIL_BTN_LABEL = 'Текст кнопки';
const EMPTY_DETAIL_BTN_LABEL = 'Текст не задан';
const DETAIL_BTN_URL = 'Ссылка для кнопки';
const EMPTY_DETAIL_BTN_URL = 'Нет ссылки';
const URL_SOURCE_PROMO_CAMPAIGN_LABEL = 'Промо-кампания';
const BEHAVIOR_TYPE_LABEL = 'Отображать QR-код';
const SHOW_VIDEO_TOUR_LABEL = 'Отображать видеоэкскурсию';

const STATUS_TYPE = {
    ACTIVE: 'Активная',
    INACTIVE: 'Неактивная',
};

const StepInfo: React.FC<IStepInfoProps> = ({ promoCampaign }) => {
    const [clientApp, setClientApp] = useState<ClientAppDto>();
    const appCode = getAppCode() || '';

    useEffect(() => {
        (async () => {
            try {
                const app = await getClientAppInfo(appCode);
                setClientApp(app);
            } catch (e) {
                console.warn(e?.message);
            }
        })();
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);

    return (
        <div className={style.containerStep}>
            <div className={style.container}>
                <Row gutter={[24, 40]}>
                    <Col span={16}>
                        <div className={style.infoTitle}>{CATEGORY}</div>
                        <div className={style.infoText}>
                            {promoCampaign.categoryList?.length
                                ? promoCampaign.categoryList.map(({ categoryName }) => categoryName).join(', ')
                                : <i>{NO_CATEGORY_LABEL}</i>
                            }
                        </div>
                    </Col>

                    <Col span={8} className={style.switchContainer}>
                        <div className={style.switch}>
                            <div className={style.infoTitle}>
                                {SHOW_VIDEO_TOUR_LABEL}
                            </div>
                            <Switch
                                checked={!promoCampaign.settings?.disabled_banner_types?.includes(BANNER_TYPE.VIDEO)}
                                disabled
                            />
                        </div>
                    </Col>

                    <Col span={8}>
                        <div className={style.infoTitle}>{URL_PROMO_CAMPAIGN}</div>
                        <div className={style.webLink}>
                            {promoCampaign.webUrl || <i>{EMPTY_URL_PROMO_CAMPAIGN}</i>}
                        </div>
                    </Col>

                    <Col span={8}>
                        <div className={style.infoTitle}>{URL_SOURCE_LABEL}</div>
                        <div className={style.flexRow}>
                            <Radio
                                className={style.checkbox}
                                checked={promoCampaign?.settings?.priority_on_web_url !== true}
                                disabled
                            >
                                {URL_SOURCE_DZO_LABEL}
                            </Radio>
                            <Radio
                                className={style.checkbox}
                                checked={promoCampaign?.settings?.priority_on_web_url === true}
                                disabled
                            >
                                {URL_SOURCE_PROMO_CAMPAIGN_LABEL}
                            </Radio>
                        </div>
                    </Col>

                    <Col span={8} className={style.switchContainer}>
                        <div className={style.switch}>
                            <div className={style.infoTitle}>{SHOW_GO_TO_LINK_LABEL}</div>
                            <Switch
                                disabled
                                checked={promoCampaign.settings?.alternative_offer_mechanic || false}
                            />
                        </div>
                    </Col>

                    <Col span={8}>
                        <div className={style.infoTitle}>{DZO}</div>
                        <div className={style.infoText}>
                            {promoCampaign.dzoName}
                        </div>
                    </Col>

                    <Col span={8}>
                        <div className={style.infoTitle}>{ACTIVE_PERIOD}</div>
                        <div className={style.infoText}>
                            {promoCampaign.startDate} - {promoCampaign.finishDate}
                        </div>
                    </Col>

                    <Col span={8} className={style.switchContainer}>
                        <div className={style.switch}>
                            <div className={style.infoTitle}>{SHOW_ONLY_IN_BUNDLE}</div>
                            <Switch disabled checked={!promoCampaign.standalone} />
                        </div>
                    </Col>

                    <Col span={8}>
                        <div className={style.infoTitle}>{TYPE_PROMO_CODE}</div>
                        <div className={style.infoText}>
                            {promoCodeTypes[promoCampaign.promoCodeType].label}
                        </div>
                    </Col>

                    <Col span={8}>
                        <div className={style.infoTitle}>{EXTERNAL_ID_LABEL}</div>
                        <div className={style.infoText}>
                            {promoCampaign.externalId || <i>{EMPTY_EXTERNAL_ID_LABEL}</i>}
                        </div>
                    </Col>

                    <Col span={8}>
                        <div className={style.infoTitle}>{STATUS}</div>
                        <div className={cn(style.status, { [style.danger]: !promoCampaign.active })}>
                            {promoCampaign.active ? STATUS_TYPE.ACTIVE : STATUS_TYPE.INACTIVE}
                        </div>
                    </Col>

                    <Col span={8}>
                        <div className={style.infoTitle}>{BEHAVIOR_TYPE_LABEL}</div>
                        <div className={style.infoText}>
                            <Switch
                                disabled
                                checked={promoCampaign.behaviorType === behaviorTypes.QR}
                            />
                        </div>
                    </Col>

                    <Col span={16} className={style.formGroup}>
                        <span className={style.formGroupLabel}>
                            Настройки детальной кнопки
                        </span>
                        <Col span={12}>
                            <div className={style.infoTitle}>{DETAIL_BTN_LABEL}</div>
                            <div className={style.infoText}>
                                {promoCampaign.settings.details_button_label || <i>{EMPTY_DETAIL_BTN_LABEL}</i>}
                            </div>
                        </Col>
                        <Col span={12}>
                            <div className={style.infoTitle}>{DETAIL_BTN_URL}</div>
                            <div className={style.infoText}>
                                {promoCampaign.settings.details_button_url || <i>{EMPTY_DETAIL_BTN_URL}</i>}
                            </div>
                        </Col>
                    </Col>

                </Row>
            </div>

            <div className={style.infoDetail}>
                <Row gutter={[16, 16]}>
                    <Col span={12}>
                        <div className={style.container}>
                            <div className={style.infoTitle}>{SHOW_PROMO_CAMPAIGN}</div>
                            <div className={style.infoText}>
                                {clientApp?.displayName}
                            </div>
                        </div>
                    </Col>
                    <Col span={12}>
                        <div className={style.container}>
                            <div className={style.infoTitle}>{TYPE_PROMO_CAMPAIGN}</div>
                            <div className={style.flex}>
                                <Radio
                                    className={style.checkbox}
                                    checked={promoCampaign.type === 'NORMAL'}
                                    disabled
                                >
                                    {EXCURSION}
                                </Radio>
                                <Radio
                                    className={style.checkbox}
                                    checked={promoCampaign.type === 'PRESENT'}
                                    disabled
                                >
                                    {GIFT}
                                </Radio>
                            </div>
                        </div>
                    </Col>
                </Row>
            </div>
        </div>
    );
};

export default StepInfo;
