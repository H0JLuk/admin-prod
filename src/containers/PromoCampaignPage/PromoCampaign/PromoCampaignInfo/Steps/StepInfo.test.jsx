import React from 'react';
import StepInfo from './StepInfo';
import { mount } from 'enzyme';
import * as clientAppService from '../../../../../api/services/clientAppService';
import { clientAppTestData, promoCampaignTestData } from '../../../../../../__tests__/constants';
import { sleep } from '../../../../../setupTests';

const NO_CATEGORY_LABEL = 'Нет выбранных категорий';
const EMPTY_URL_PROMO_CAMPAIGN = 'Нет ссылки';

describe('<StepInfo/> test', () => {
    const props = {
        promoCampaign: promoCampaignTestData,
    };

    beforeEach(() => {
        clientAppService.getClientAppInfo = jest.fn().mockResolvedValue(clientAppTestData);
    });

    it('should match to snapshot', async () => {
        const wrapper = mount(<StepInfo { ...props } />);
        await sleep();

        expect(wrapper.debug()).toMatchSnapshot();
    });

    it('should call `getClientAppInfo` function', async () => {
        mount(<StepInfo { ...props } />);
        await sleep();
        expect(clientAppService.getClientAppInfo).toHaveBeenCalledTimes(1);
    });

    it('should render label if category list is empty', async () => {
        const props = {
            promoCampaign: {
                ...promoCampaignTestData,
                categoryList: [],
            },
        };
        const wrapper = mount(<StepInfo { ...props } />);
        await sleep();
        const label = wrapper.find('.infoText').first().text();
        expect(label).toBe(NO_CATEGORY_LABEL);
    });

    it('should render label if no webUrl', async () => {
        const props = {
            promoCampaign: {
                ...promoCampaignTestData,
                webUrl: undefined,
            },
        };
        const wrapper = mount(<StepInfo { ...props } />);
        await sleep();
        const label = wrapper.find({ children: EMPTY_URL_PROMO_CAMPAIGN }).first().text();
        expect(label).toBe(EMPTY_URL_PROMO_CAMPAIGN);
    });
});
