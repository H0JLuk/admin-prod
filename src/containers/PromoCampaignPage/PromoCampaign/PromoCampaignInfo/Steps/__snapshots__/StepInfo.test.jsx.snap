// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StepInfo/> test should match to snapshot 1`] = `
"<StepInfo promoCampaign={{...}}>
  <div className=\\"containerStep\\">
    <Memo(ContentBlock)>
      <div className=\\"contentContainer\\">
        <div style={{...}}>
          <Row gutter={{...}}>
            <div className=\\"ant-row\\" style={{...}}>
              <Col span={16}>
                <div style={{...}} className=\\"ant-col ant-col-16\\">
                  <div className=\\"infoTitle\\">
                    Категории
                  </div>
                  <div className=\\"infoText\\">
                    Test category Name 1, Test category Name 2
                  </div>
                </div>
              </Col>
              <Col span={8} className=\\"switchContainer\\">
                <div style={{...}} className=\\"ant-col ant-col-8 switchContainer\\">
                  <div className=\\"switch\\">
                    <div className=\\"infoTitle\\">
                      Отображать видеоэкскурсию
                    </div>
                    <Switch checked={true} disabled={true}>
                      <Wave insertExtraNode={true}>
                        <Switch checked={true} prefixCls=\\"ant-switch\\" className=\\"\\" disabled={true} loadingIcon={{...}}>
                          <button type=\\"button\\" role=\\"switch\\" aria-checked={true} disabled={true} className=\\"ant-switch ant-switch-checked ant-switch-disabled\\" onKeyDown={[Function: onInternalKeyDown]} onClick={[Function: onInternalClick]}>
                            <div className=\\"ant-switch-handle\\" />
                            <span className=\\"ant-switch-inner\\" />
                          </button>
                        </Switch>
                      </Wave>
                    </Switch>
                  </div>
                </div>
              </Col>
              <Col span={8}>
                <div style={{...}} className=\\"ant-col ant-col-8\\">
                  <div className=\\"infoTitle\\">
                    Ссылка на страницу промо-кампании
                  </div>
                  <div className=\\"webLink\\">
                    https://www.promo.sberhealth.ru/vvo
                  </div>
                </div>
              </Col>
              <Col span={8}>
                <div style={{...}} className=\\"ant-col ant-col-8\\">
                  <div className=\\"infoTitle\\">
                    Источник ссылки
                  </div>
                  <div className=\\"flexRow\\">
                    <Radio className=\\"checkbox\\" checked={false} disabled={true} type=\\"radio\\">
                      <label className=\\"ant-radio-wrapper ant-radio-wrapper-disabled checkbox\\" style={[undefined]} onMouseEnter={[undefined]} onMouseLeave={[undefined]}>
                        <Checkbox checked={false} disabled={true} type=\\"radio\\" prefixCls=\\"ant-radio\\" className=\\"\\" style={{...}} defaultChecked={false} onFocus={[Function: onFocus]} onBlur={[Function: onBlur]} onChange={[Function: onChange]} onKeyDown={[Function: onKeyDown]} onKeyPress={[Function: onKeyPress]} onKeyUp={[Function: onKeyUp]}>
                          <span className=\\"ant-radio ant-radio-disabled\\" style={{...}}>
                            <input name={[undefined]} id={[undefined]} type=\\"radio\\" required={[undefined]} readOnly={[undefined]} disabled={true} tabIndex={[undefined]} className=\\"ant-radio-input\\" checked={false} onClick={[undefined]} onFocus={[Function: onFocus]} onBlur={[Function: onBlur]} onKeyUp={[Function: onKeyUp]} onKeyDown={[Function: onKeyDown]} onKeyPress={[Function: onKeyPress]} onChange={[Function (anonymous)]} autoFocus={[undefined]} value={[undefined]} />
                            <span className=\\"ant-radio-inner\\" />
                          </span>
                        </Checkbox>
                        <span>
                          ДЗО
                        </span>
                      </label>
                    </Radio>
                    <Radio className=\\"checkbox\\" checked={true} disabled={true} type=\\"radio\\">
                      <label className=\\"ant-radio-wrapper ant-radio-wrapper-checked ant-radio-wrapper-disabled checkbox\\" style={[undefined]} onMouseEnter={[undefined]} onMouseLeave={[undefined]}>
                        <Checkbox checked={true} disabled={true} type=\\"radio\\" prefixCls=\\"ant-radio\\" className=\\"\\" style={{...}} defaultChecked={false} onFocus={[Function: onFocus]} onBlur={[Function: onBlur]} onChange={[Function: onChange]} onKeyDown={[Function: onKeyDown]} onKeyPress={[Function: onKeyPress]} onKeyUp={[Function: onKeyUp]}>
                          <span className=\\"ant-radio ant-radio-checked ant-radio-disabled\\" style={{...}}>
                            <input name={[undefined]} id={[undefined]} type=\\"radio\\" required={[undefined]} readOnly={[undefined]} disabled={true} tabIndex={[undefined]} className=\\"ant-radio-input\\" checked={true} onClick={[undefined]} onFocus={[Function: onFocus]} onBlur={[Function: onBlur]} onKeyUp={[Function: onKeyUp]} onKeyDown={[Function: onKeyDown]} onKeyPress={[Function: onKeyPress]} onChange={[Function (anonymous)]} autoFocus={[undefined]} value={[undefined]} />
                            <span className=\\"ant-radio-inner\\" />
                          </span>
                        </Checkbox>
                        <span>
                          Промо-кампания
                        </span>
                      </label>
                    </Radio>
                  </div>
                </div>
              </Col>
              <Col span={8} className=\\"switchContainer\\">
                <div style={{...}} className=\\"ant-col ant-col-8 switchContainer\\">
                  <div className=\\"switch\\">
                    <div className=\\"infoTitle\\">
                      Отображать кнопку &quot;Перейти на сайт&quot;
                    </div>
                    <Switch disabled={true} checked={false}>
                      <Wave insertExtraNode={true}>
                        <Switch checked={false} prefixCls=\\"ant-switch\\" className=\\"\\" disabled={true} loadingIcon={{...}}>
                          <button type=\\"button\\" role=\\"switch\\" aria-checked={false} disabled={true} className=\\"ant-switch ant-switch-disabled\\" onKeyDown={[Function: onInternalKeyDown]} onClick={[Function: onInternalClick]}>
                            <div className=\\"ant-switch-handle\\" />
                            <span className=\\"ant-switch-inner\\" />
                          </button>
                        </Switch>
                      </Wave>
                    </Switch>
                  </div>
                </div>
              </Col>
              <Col span={8}>
                <div style={{...}} className=\\"ant-col ant-col-8\\">
                  <div className=\\"infoTitle\\">
                    ДЗО
                  </div>
                  <div className=\\"infoText\\">
                    СберЗдоровье
                  </div>
                </div>
              </Col>
              <Col span={8}>
                <div style={{...}} className=\\"ant-col ant-col-8\\">
                  <div className=\\"infoTitle\\">
                    Период действия
                  </div>
                  <div className=\\"infoText\\">
                    2021-02-12T21:00:00
                     - 
                    2021-03-08T20:59:59.999
                  </div>
                </div>
              </Col>
              <Col span={8} className=\\"switchContainer\\">
                <div style={{...}} className=\\"ant-col ant-col-8 switchContainer\\">
                  <div className=\\"switch\\">
                    <div className=\\"infoTitle\\">
                      Отображать только в составе бандла
                    </div>
                    <Switch disabled={true} checked={false}>
                      <Wave insertExtraNode={true}>
                        <Switch checked={false} prefixCls=\\"ant-switch\\" className=\\"\\" disabled={true} loadingIcon={{...}}>
                          <button type=\\"button\\" role=\\"switch\\" aria-checked={false} disabled={true} className=\\"ant-switch ant-switch-disabled\\" onKeyDown={[Function: onInternalKeyDown]} onClick={[Function: onInternalClick]}>
                            <div className=\\"ant-switch-handle\\" />
                            <span className=\\"ant-switch-inner\\" />
                          </button>
                        </Switch>
                      </Wave>
                    </Switch>
                  </div>
                </div>
              </Col>
              <Col span={8}>
                <div style={{...}} className=\\"ant-col ant-col-8\\">
                  <div className=\\"infoTitle\\">
                    Тип промокода
                  </div>
                  <div className=\\"infoText\\">
                    Персональный
                  </div>
                </div>
              </Col>
              <Col span={8}>
                <div style={{...}} className=\\"ant-col ant-col-8\\">
                  <div className=\\"infoTitle\\">
                    Внешний ID
                  </div>
                  <div className=\\"infoText\\">
                    <i>
                      Внешний ID не выбран
                    </i>
                  </div>
                </div>
              </Col>
              <Col span={8}>
                <div style={{...}} className=\\"ant-col ant-col-8\\">
                  <div className=\\"infoTitle\\">
                    Активность промо-кампании
                  </div>
                  <div className=\\"status\\">
                    Активная
                  </div>
                </div>
              </Col>
              <Col span={8}>
                <div style={{...}} className=\\"ant-col ant-col-8\\">
                  <div className=\\"infoTitle\\">
                    Отображать QR-код
                  </div>
                  <div className=\\"infoText\\">
                    <Switch disabled={true} checked={true}>
                      <Wave insertExtraNode={true}>
                        <Switch checked={true} prefixCls=\\"ant-switch\\" className=\\"\\" disabled={true} loadingIcon={{...}}>
                          <button type=\\"button\\" role=\\"switch\\" aria-checked={true} disabled={true} className=\\"ant-switch ant-switch-checked ant-switch-disabled\\" onKeyDown={[Function: onInternalKeyDown]} onClick={[Function: onInternalClick]}>
                            <div className=\\"ant-switch-handle\\" />
                            <span className=\\"ant-switch-inner\\" />
                          </button>
                        </Switch>
                      </Wave>
                    </Switch>
                  </div>
                </div>
              </Col>
              <Col span={8}>
                <div style={{...}} className=\\"ant-col ant-col-8\\">
                  <div className=\\"infoTitle\\">
                    Оформить продажу
                  </div>
                  <div className=\\"infoText\\">
                    <Switch disabled={true} checked={false}>
                      <Wave insertExtraNode={true}>
                        <Switch checked={false} prefixCls=\\"ant-switch\\" className=\\"\\" disabled={true} loadingIcon={{...}}>
                          <button type=\\"button\\" role=\\"switch\\" aria-checked={false} disabled={true} className=\\"ant-switch ant-switch-disabled\\" onKeyDown={[Function: onInternalKeyDown]} onClick={[Function: onInternalClick]}>
                            <div className=\\"ant-switch-handle\\" />
                            <span className=\\"ant-switch-inner\\" />
                          </button>
                        </Switch>
                      </Wave>
                    </Switch>
                  </div>
                </div>
              </Col>
              <Col span={16} className=\\"formGroup\\">
                <div style={{...}} className=\\"ant-col ant-col-16 formGroup\\">
                  <span className=\\"formGroupLabel\\">
                    Настройки детальной кнопки
                  </span>
                  <Col span={12}>
                    <div style={{...}} className=\\"ant-col ant-col-12\\">
                      <div className=\\"infoTitle\\">
                        Текст кнопки
                      </div>
                      <div className=\\"infoText\\">
                        <i>
                          Текст не задан
                        </i>
                      </div>
                    </div>
                  </Col>
                  <Col span={12}>
                    <div style={{...}} className=\\"ant-col ant-col-12\\">
                      <div className=\\"infoTitle\\">
                        Ссылка для кнопки
                      </div>
                      <div className=\\"infoText\\">
                        <i>
                          Нет ссылки
                        </i>
                      </div>
                    </div>
                  </Col>
                </div>
              </Col>
            </div>
          </Row>
        </div>
      </div>
    </Memo(ContentBlock)>
    <div className=\\"infoDetail\\">
      <Row gutter={{...}}>
        <div className=\\"ant-row\\" style={{...}}>
          <Col span={12}>
            <div style={{...}} className=\\"ant-col ant-col-12\\">
              <Memo(ContentBlock)>
                <div className=\\"contentContainer\\">
                  <div style={{...}}>
                    <div className=\\"infoTitle\\">
                      Витрина, в которой отображается промо-кампания
                    </div>
                    <div className=\\"infoText\\" />
                  </div>
                </div>
              </Memo(ContentBlock)>
            </div>
          </Col>
          <Col span={12}>
            <div style={{...}} className=\\"ant-col ant-col-12\\">
              <Memo(ContentBlock)>
                <div className=\\"contentContainer\\">
                  <div style={{...}}>
                    <div className=\\"infoTitle\\">
                      Тип промо-кампании
                    </div>
                    <div className=\\"flex\\">
                      <Radio className=\\"checkbox\\" checked={false} disabled={true} type=\\"radio\\">
                        <label className=\\"ant-radio-wrapper ant-radio-wrapper-disabled checkbox\\" style={[undefined]} onMouseEnter={[undefined]} onMouseLeave={[undefined]}>
                          <Checkbox checked={false} disabled={true} type=\\"radio\\" prefixCls=\\"ant-radio\\" className=\\"\\" style={{...}} defaultChecked={false} onFocus={[Function: onFocus]} onBlur={[Function: onBlur]} onChange={[Function: onChange]} onKeyDown={[Function: onKeyDown]} onKeyPress={[Function: onKeyPress]} onKeyUp={[Function: onKeyUp]}>
                            <span className=\\"ant-radio ant-radio-disabled\\" style={{...}}>
                              <input name={[undefined]} id={[undefined]} type=\\"radio\\" required={[undefined]} readOnly={[undefined]} disabled={true} tabIndex={[undefined]} className=\\"ant-radio-input\\" checked={false} onClick={[undefined]} onFocus={[Function: onFocus]} onBlur={[Function: onBlur]} onKeyUp={[Function: onKeyUp]} onKeyDown={[Function: onKeyDown]} onKeyPress={[Function: onKeyPress]} onChange={[Function (anonymous)]} autoFocus={[undefined]} value={[undefined]} />
                              <span className=\\"ant-radio-inner\\" />
                            </span>
                          </Checkbox>
                          <span>
                            Экскурсия
                          </span>
                        </label>
                      </Radio>
                      <Radio className=\\"checkbox\\" checked={true} disabled={true} type=\\"radio\\">
                        <label className=\\"ant-radio-wrapper ant-radio-wrapper-checked ant-radio-wrapper-disabled checkbox\\" style={[undefined]} onMouseEnter={[undefined]} onMouseLeave={[undefined]}>
                          <Checkbox checked={true} disabled={true} type=\\"radio\\" prefixCls=\\"ant-radio\\" className=\\"\\" style={{...}} defaultChecked={false} onFocus={[Function: onFocus]} onBlur={[Function: onBlur]} onChange={[Function: onChange]} onKeyDown={[Function: onKeyDown]} onKeyPress={[Function: onKeyPress]} onKeyUp={[Function: onKeyUp]}>
                            <span className=\\"ant-radio ant-radio-checked ant-radio-disabled\\" style={{...}}>
                              <input name={[undefined]} id={[undefined]} type=\\"radio\\" required={[undefined]} readOnly={[undefined]} disabled={true} tabIndex={[undefined]} className=\\"ant-radio-input\\" checked={true} onClick={[undefined]} onFocus={[Function: onFocus]} onBlur={[Function: onBlur]} onKeyUp={[Function: onKeyUp]} onKeyDown={[Function: onKeyDown]} onKeyPress={[Function: onKeyPress]} onChange={[Function (anonymous)]} autoFocus={[undefined]} value={[undefined]} />
                              <span className=\\"ant-radio-inner\\" />
                            </span>
                          </Checkbox>
                          <span>
                            Подарок
                          </span>
                        </label>
                      </Radio>
                    </div>
                  </div>
                </div>
              </Memo(ContentBlock)>
            </div>
          </Col>
        </div>
      </Row>
    </div>
  </div>
</StepInfo>"
`;
